### PR TITLE
add logger

### PR DESCRIPTION
**Pull request by superstar54**
There are too many outputs when running Batoms. e.g.

```
set_attributes second_bond: 0.0002613067626953125
Build geometry nodes for bonds: 0.008374452590942383
bonds: build_object:       0.01 s
Build bondlist: 0.00
calc_bond_data:       0.00 s
set_attributes atoms_index1: 2.4557113647460938e-05
set_attributes atoms_index2: 5.698204040527344e-05
```

The information is useful for debugging, but not relevant to the user.

Solution
=================
The following solutions are implemented: 

- add a logger to print the information in the command line and save the information into a log file.
- using `self.report` to print information inside Blender.

User can set the logging level in the preference:

System information is also saved to the log file (/tmp/beautiful_atoms.log in Linux), e.g.
```
INFO     [set_logger          ]: Blender version: 3.0.0 
INFO     [set_logger          ]: Python version: 3.9.7 (default, Oct 11 2021, 10:06:01)  [GCC 9.3.1 20200408 (Red Hat 9.3.1-2)] 
INFO     [set_logger          ]: Beautiful Atoms version: (2, 0, 0) 
INFO     [set_logger          ]: Beautiful Atoms directory: /home/xing/apps/beautiful-atoms
```
This will be helpful when a user has a problem.

### CODE REVIEW

**Review by alchem0x2A — COMMENTED** [review]
LGTM. just a minor comment on the scope of `getLogger`

File: batoms/attribute.py [hunk 1]
```diff
@@ -5,6 +5,8 @@
 import numpy as np
 from time import time
 from batoms.base.collection import Setting
+import logging
+logger = logging.getLogger('batoms')
```
[the remark this answers is not in the document]
Comment (superstar54) [reply batoms/attribute.py:9]: It's strange that I tried `logging.getLogger(__name__)`, but it does not work. Instead, each submodule gets its own logger and  I can't set the logging level for all of them.

File: batoms/logger.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+
+# logger = logging.getLogger('batoms')
+logger = logging.getLogger(__name__)
+root_logger = logging.getLogger("batoms")
+
+def set_logger(version):
+    # formatter = ('%(levelname)-8s '
+    #                             '[%(funcName)-20s]: %(message)s')
+    formatter = ('%(levelname)s '
```
Comment (alchem0x2A) [inline batoms/logger.py:22]: @superstar54 I modified the logger output so module & func name are both shown. Please modify as you prefer

File: batoms/attribute.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (superstar54) [reply batoms/attribute.py]: :+1: 
Yes, it's ok to resolve.

**Pull request merged**